helm/charts/hydra/templates/janitor-cron-job.yaml

```diff
@@ -0,0 +1,103 @@
+{{- if .Values.janitor.enabled -}}
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: {{ include "hydra.fullname" . }}-janitor
+  {{- if .Release.Namespace }}
+  namespace: {{ .Release.Namespace }}
+  {{- end }}
+  labels:
+    {{- include "hydra.janitor.labels" . | nindent 4 }}
+    {{- with .Values.janitor.cronjob.labels }}
+      {{- toYaml . | nindent 4 }}
+    {{- end }}
+  annotations:
+    {{- with .Values.janitor.cronjob.annotations }}
+      {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  concurrencyPolicy: Forbid
+  schedule: {{ .Values.janitor.cronjob.schedule | quote }}
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          labels:
+            {{- include "hydra.janitor.labels" . | nindent 12 }}
+            {{- with .Values.janitor.cronjob.labels }}
+              {{- toYaml . | nindent 12 }}
+            {{- end }}
+            {{- with .Values.janitor.cronjob.podMetadata.labels }}
+              {{- toYaml . | nindent 12 }}
+            {{- end }}
+          annotations:
+            {{- include "hydra.annotations.checksum" . | nindent 12 -}}
+            {{- with .Values.janitor.cronjob.annotations }}
+              {{- toYaml . | nindent 12 }}
+            {{- end }}
+            {{- with $.Values.janitor.cronjob.podMetadata.annotations }}
+              {{- toYaml . | nindent 12 }}
+            {{- end }}
+        spec:
+          restartPolicy: OnFailure
+          {{- with .Values.imagePullSecrets }}
+          imagePullSecrets:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          volumes:
+            - name: {{ include "hydra.name" . }}-config-volume
+              configMap:
+                name: {{ include "hydra.fullname" . }}
+          containers:
+            - name: janitor
+              image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+              imagePullPolicy: {{ .Values.image.pullPolicy }}
+              {{- if .Values.janitor.cronjob.securityContext.enabled }}
+              securityContext:
+                {{- omit .Values.janitor.cronjob.securityContext "enabled" | toYaml | nindent 16 }}
+              {{- end }}
```

Comment:
I would replace the block with a simple `{{- with .Values.janitor.cronjob.securityContext }}`, which won't render if the object is empty. Unless we want to specify some defaultContext, in which case we need to update the values.yaml 😉

Reply:
>Unless we want to specify some defaultContext, in which case we need to update the values.yaml

Do you think we should? I see that the deployment has a default value configured, but I don't know what the best practice is for Helm charts out in the world.

Reply:
I'd say, since we control the binary and how we deploy, we might want to include default Context. We know the required permissions :) IMHO, we should set them as the same as we have on the deployment